engine/src/main/java/org/terasology/rendering/dag/nodes/InitialPostProcessingNode.java

```diff
@@ -100,11 +100,15 @@ public InitialPostProcessingNode(Context context) {
         lightShaftsAreEnabled = renderingConfig.isLightShafts();
         renderingConfig.subscribe(RenderingConfig.LIGHT_SHAFTS, this);
 
+        // TODO: Temporary hack for now.
+        FBOConfig one8thScaleBloomConfig = new FBOConfig(BloomBlurNode.ONE_8TH_SCALE_FBO, ONE_8TH_SCALE, FBO.Type.DEFAULT);
+        FBO one8thBloomFbo = requiresFBO(one8thScaleBloomConfig, displayResolutionDependentFBOs);
+
         int textureSlot = 0;
-        addDesiredStateChange(new SetInputTextureFromFbo(textureSlot++, READONLY_GBUFFER, ColorTexture, displayResolutionDependentFBOs, INITIAL_POST_MATERIAL, "texScene"));
+        addDesiredStateChange(new SetInputTextureFromFbo(textureSlot++, displayResolutionDependentFBOs.getPrimaryBuffer(), ColorTexture, displayResolutionDependentFBOs, INITIAL_POST_MATERIAL, "texScene"));
         addDesiredStateChange(new SetInputTexture(textureSlot++, "engine:vignette", INITIAL_POST_MATERIAL, "texVignette"));
-        setBloomInputTexture = new SetInputTextureFromFbo(textureSlot++, ONE_8TH_SCALE_FBO, ColorTexture, displayResolutionDependentFBOs, INITIAL_POST_MATERIAL, "texBloom");
-        setLightShaftsInputTexture = new SetInputTextureFromFbo(textureSlot, LIGHT_SHAFTS_FBO, ColorTexture, displayResolutionDependentFBOs, INITIAL_POST_MATERIAL, "texLightShafts");
+        setBloomInputTexture = new SetInputTextureFromFbo(textureSlot++, one8thBloomFbo, ColorTexture, displayResolutionDependentFBOs, INITIAL_POST_MATERIAL, "texBloom");
```

Comment:
ONE_8TH_SCALE_FBO is the urn of an FBO, and one8thBloomFbo is the fbo corresponding to that urn. No scaling factors involved anywhere. However, the change (from urn to fbo) is because I modified all stateChanges to accept raw FBOs (and not URBs) coz it's much cleaner (and more efficient).